src/cpp/rediscluster.cpp

```diff
@@ -53,7 +53,16 @@ RedisCluster::~RedisCluster()
 
 CommandReply RedisCluster::run(Command& cmd)
 {
-    std::string db_prefix = this->_get_db_node_prefix(cmd);
+    std::string db_prefix;
+
+    if (cmd.has_keys())
+        db_prefix = this->_get_db_node_prefix(cmd);
+    else if (this->is_addressable(cmd.get_address(), cmd.get_port()))
```

Comment:
I wonder if we should put the address, port case first so that if an address and port are specified, it always overrides the presence of keys.